### DISCUSSION TITLE
ı learn python and help qtile but?

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#3103] [Q&A]
where is libqtile and qtile how works ı love qtile this is not boring
maybe forward ı make my own qtile ? for learning

**Comment by jwijenbergh** [comment]
Sorry, but is there a specific question here? :sweat_smile: 

Learning how to make your own window manager is a good process for learning. But Qtile is so configurable you can do a lot with it.

**Reply by m-col** [reply]
> Learning how to make your own window manager is a good process for learning.

Yeah definitely - but it is no small task.

Also consider that learning how to make a window manager and learning Python are different tasks with their own issues. Not that I'm suggesting you don't, it's just something to bear in mind!

> where is libqtile

You can download Qtile with: `git clone https://github.com/qtile/qtile`. This will make a new folder, and inside that folder is another folder called `libqtile.` This is the source code. The `qtile` command is available at `./bin/qtile`, so if you start from there you can follow the flow of the program to see how everything is set up and started.

**Reply by jwijenbergh** [reply]
> Yeah definitely - but it is no small task. Also consider that learning how to make a window manager and learning Python are different tasks with their own issues. Not that I'm suggesting you don't, it's just something to bear in mind!

Can confirm :sweat_smile:
